src/reader/reader.c

```diff
@@ -0,0 +1,35 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   reader.c                                           :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: lporoshi <lporoshi@student.42.fr>          +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/12/09 15:15:07 by lporoshi          #+#    #+#             */
+/*   Updated: 2023/12/09 17:06:18 by lporoshi         ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "../../include/reader.h"
+#include "../../config.h"
+
+void	init_reader(void)
+{
+	using_history();
+}
+
+void	close_reader(void)
+{
+	rl_clear_history();
+}
+
+char	*get_line(void)
+{
+	char			*line;
+
+	line = readline(SHELL_PROMPT);
+	if (line == NULL || line[0] == '\0')
+		return (NULL);
+	add_history(line);
+	return (line);
+}
```

Comment:
how are we going to use history in the shell? if we gonna handle arrow-up to iterate through the history or 'history' linux command, we might need additional functions like getting a last command from the history.

Reply:
we dont need to do it